### PR TITLE
Storing and Retrieving Audio Success

### PR DESCRIPTION
**Pull request by Axelh04**
Successfully added playback using GET requests to my database to fetch a byteArray that will convert into a blob. Fixed the formatting of the code, added modularity.

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -7,6 +7,13 @@ const prisma = new PrismaClient()
 const session = require('express-session');
 const cors = require('cors');
 const userRoutes = require('./users.js');
+const audioRoutes = require('./audios.js');
+
+
+const bodyParser = require('body-parser'); 
+
+app.use(bodyParser.json({ limit: '50mb' }));
+app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }));
```
Comment (kfullen) [inline backend/index.js:16]: Does this allow you to process the audio data in the request if the data is large?

File: backend/audios.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const express = require('express');
```
Comment (kfullen) [inline backend/audios.js:1]: Really love how you're organizing your file structure with having routes for each schema correspond to different files! Makes it very clear to understand

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
-import { UserContext } from './UserContext';
-import { BrowserRouter, Routes, Route } from 'react-router-dom';
-import Landing from './Landing';
-import LoginForm from './LoginForm';
-import SignupForm from './SignupForm';
-import Temp from './Temp';
+import { useState, useEffect } from "react";
+import { UserContext } from "./UserContext";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import Landing from "./LandingPageComp/Landing";
```
Comment (kfullen) [inline frontend/src/App.jsx:4]: Also love the file organization in the frontend!

File: frontend/src/UserAuth/LoginForm/LoginForm.jsx
Comment (nhsoukayna) [inline frontend/src/UserAuth/LoginForm/LoginForm.jsx:1]: This is unrelated to storing and receiving audio.
I suggest to have this change in a separate commit/PR.

File: frontend/src/UserAuth/SignupForm/SignupForm.css [hunk 4]
```diff
@@ -0,0 +1,50 @@
+.signup-form-container {
```
Comment (nhsoukayna) [inline frontend/src/UserAuth/SignupForm/SignupForm.css:1]: Similarly since this is for signup also did not need to be part of this PR.

File: frontend/src/UserAuth/SignupForm/SignupForm.jsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
-import { useState, useContext } from 'react';
```
Comment (nhsoukayna) [inline frontend/src/UserAuth/SignupForm/SignupForm.jsx:1]: Same thing here this change would be better split out since unrelated to audio operations.

File: backend/index.js (hunk 1 above)
Comment (nhsoukayna) [reply backend/index.js:16]: +1. Also why did you chose 50MB?

File: frontend/src/RecordingPageComp/RecordingComp.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  const recorderControls = useAudioRecorder();
+  const reader = new FileReader();
+
+  const addAudioElement = (blob) => {
+    reader.readAsDataURL(blob);
+    const src = URL.createObjectURL(blob);
+    setAudioSource(src);
+
+    reader.onload = function (event) {
+      const audiodata = event.target.result.split(",")[1]; //Stores part of URL after the ,
```
Comment (nhsoukayna) [inline frontend/src/RecordingPageComp/RecordingComp.jsx:17]: It would be safest to add some error handling if there is no "," in result. 
Currently it would throw index out of bound.

File: backend/audios.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+const router = express.Router();
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+
+
+//Save audio recording to DB 
+router.post("/users/:id/audios/create", async (req, res) => {
```
Comment (nhsoukayna) [inline backend/audios.js:8]: It would be helpful to add typing to your function parameters.
This would allow you catch type errors before running your code.
In js types are only known at runtime and so unless you add the types you would not be able to tell if there is an error in your typing until you run your code. 
Example:

```
router.post("/users/:id/audios/create", async (req: Request, res: Response) => {
```

**Review by nhsoukayna — COMMENTED** [review]
Great progress Axel! 

I added inline comments but my feedback is mainly around:
(1) Limit PR scope: Going forward it would be helpful to limit the scope of the PR to a single change (even things like moving folders or renaming could be done in separate PR). This would make the review easier. 
(2) Use type checking: This would help you catch type errors earlier in development cycle.

File: frontend/src/RecordingPageComp/RecordingComp.jsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    // If doesn't exist, create it and append it to the body
+    if (!audio) {
+      audio = document.createElement("audio");
+      audio.id = "myAudioPlayer"; // Set an ID for easy retrieval
+      audio.controls = true;
+      document.body.appendChild(audio);
+    }
+
+    // update the source of the audio element
+    audio.src = audioSrc;
```
Comment (kfullen) [inline frontend/src/RecordingPageComp/RecordingComp.jsx:41]: Just noticed this now, but instead of using document methods to update the UI, which goes against the idea of using React, you should save the audio to a state variable and render audio elements below in the returned component based on the state variable (i.e. mapping audio to audio elements).
